File: frontend/demo/component/datepicker/date-picker-initial-position.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/date-picker';
import { applyTheme } from 'Frontend/generated/theme';
import { formatISO, lastDayOfYear } from 'date-fns';

@customElement('date-picker-initial-position')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  private lastDayOfTheYear = lastDayOfYear(Date.now());

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-date-picker
        label="Q4 deadline"
        .initialPosition="${formatISO(this.lastDayOfTheYear, { representation: 'date' })}"
      ></vaadin-date-picker>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/datepicker/DatePickerInitialPosition.java
package com.vaadin.demo.component.datepicker;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.datepicker.DatePicker;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;

import java.time.LocalDate;
import java.time.ZoneId;
import java.time.temporal.TemporalAdjusters;

@Route("date-picker-initial-position")
public class DatePickerInitialPosition extends Div {

    public DatePickerInitialPosition() {
        DatePicker datePicker = new DatePicker("Q4 deadline");
        // tag::snippet[]
        LocalDate lastDayOfYear = LocalDate.now(ZoneId.systemDefault())
                .with(TemporalAdjusters.lastDayOfYear());

        datePicker.setInitialPosition(lastDayOfYear);
        // end::snippet[]

        add(datePicker);
    }
    public static class Exporter extends DemoExporter<DatePickerInitialPosition> { // hidden-source-line
    } // hidden-source-line
}
